lib/sdf-server/src/service/v2/change_set/cancel_approval_request.rs

```diff
@@ -0,0 +1,54 @@
+use axum::extract::{Host, OriginalUri, Path};
+use dal::{ChangeSet, ChangeSetId, WorkspacePk, WsEvent};
+
+use super::{Error, Result};
+use crate::{
+    extract::{AccessBuilder, HandlerContext, PosthogClient},
+    track,
+};
+
+pub async fn cancel_approval_request(
+    HandlerContext(builder): HandlerContext,
+    AccessBuilder(request_ctx): AccessBuilder,
+    PosthogClient(posthog_client): PosthogClient,
+    OriginalUri(original_uri): OriginalUri,
+    Host(host_name): Host,
+    Path((_workspace_pk, change_set_id)): Path<(WorkspacePk, ChangeSetId)>,
+) -> Result<()> {
+    let ctx = builder
+        .build(request_ctx.build(change_set_id.into()))
+        .await?;
+
+    let mut change_set = ChangeSet::find(&ctx, ctx.visibility().change_set_id)
+        .await?
+        .ok_or(Error::ChangeSetNotFound(ctx.change_set_id()))?;
+    let old_status = change_set.status;
+
+    change_set.reopen_change_set(&ctx).await?;
+
+    track(
+        &posthog_client,
+        &ctx,
+        &original_uri,
+        &host_name,
+        "approve_change_set_apply",
+        serde_json::json!({
+            "merged_change_set": change_set_id,
+        }),
+    );
+
+    WsEvent::change_set_status_changed(
+        &ctx,
+        change_set.id,
+        ChangeSet::extract_userid_from_context(&ctx).await,
+        old_status,
+        change_set.status,
+    )
+    .await?
+    .publish_on_commit(&ctx)
```

Comment:
I think our near-future world of forwarding these along to be published by the rebaser is going to be great. I saw the `...publish_on_commit().await?;` immediately followed by `ctx.commit.await?;` and kind of rolled my eyes--even though this code is *exactly* the way it should be currently. Soooon